### ISSUE TITLE
MCP320x_read pin swapped

### ISSUE THREAD

**Issue by zfi** [parallaxinc/solo#407]
The domToMutation method appears to swap the chip_select and clock pins whereas the mutationToDom does not. 
```
  domToMutation: function(container) {
    const ch_c = container.getAttribute('chip');
    const ck_pin = container.getAttribute('cs_pin', cs_pin);
    var cs_pin = container.getAttribute('ck_pin', ck_pin);
    var do_pin = container.getAttribute('do_pin', do_pin);
    var di_pin = container.getAttribute('di_pin', di_pin);
    this.updateShape_({'CH_C': ch_c, 'CK': ck_pin, 'CS': cs_pin, 'DO': do_pin, 'DI': di_pin});
```
```
  mutationToDom: function() {
    const container = document.createElement('mutation');
    const chc = this.getFieldValue('CHIP');
    const csPin = this.getFieldValue('CS_PIN');
    const ckPin = this.getFieldValue('CLK_PIN');
    const doPin = this.getFieldValue('DO_PIN');
    const diPin = this.getFieldValue('DI_PIN');
    container.setAttribute('chip', chc);
    container.setAttribute('cs_pin', csPin);
    container.setAttribute('ck_pin', ckPin);
    container.setAttribute('do_pin', doPin);
    container.setAttribute('di_pin', diPin);
    return container;
  },
```

**Comment by zfi** [comment]
Here is the proposed refactor of the above code. It seems a lot less prone to errors like the one reported.
```
  mutationToDom: function() {
    const container = document.createElement('mutation');
    container.setAttribute('chip', this.getFieldValue('CHIP'));
    container.setAttribute('cs_pin', this.getFieldValue('CS_PIN'));
    container.setAttribute('ck_pin', this.getFieldValue('CLK_PIN'));
    container.setAttribute('do_pin', this.getFieldValue('DO_PIN'));
    container.setAttribute('di_pin', this.getFieldValue('DI_PIN'));
    return container;
  },
  domToMutation: function(container) {
    this.updateShape_({
      'CH_C': container.getAttribute('chip'),
      'CK': container.getAttribute('ck_pin'),
      'CS': container.getAttribute('cs_pin'),
      'DO': container.getAttribute('do_pin'),
      'DI': container.getAttribute('di_pin'),
    });
  },
```

**Comment by zfi** [comment]
This issue was addressed in PR #409.

**Comment by zfi** [comment]
The refactored code makes it pretty much impossible to get the pins swapped up.
Closing this issue.
